crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,14 +37,13 @@ pub struct DynamicEntity {
 
 impl DynamicScene {
     /// Create a new dynamic scene from a given scene.
-    pub fn from_scene(scene: &Scene, type_registry: &AppTypeRegistry) -> Self {
-        Self::from_world(&scene.world, type_registry)
+    pub fn from_scene(scene: &Scene) -> Self {
+        Self::from_world(&scene.world)
     }
 
     /// Create a new dynamic scene from a given world.
-    pub fn from_world(world: &World, type_registry: &AppTypeRegistry) -> Self {
-        let mut builder =
-            DynamicSceneBuilder::from_world_with_type_registry(world, type_registry.clone());
+    pub fn from_world(world: &World) -> Self {
```

Comment:
Should this be a trait implementation? We could move this into an `impl FromWorld for DynamicScene` since it doesn't need the second argument anymore.

Reply:
Good point!

Reply:
Hm, I just checked the `FromWorld` trait definition and it looks like it actually takes `&mut World`. So looks like we can't do this :/

Reply:
Maybe we could `impl From<&World> for DynamicSceneBuilder` instead?

Reply:
Possibly. I'm also starting to wonder if doing `From` impls has a negative effect on discoverability. Constructors directly on the struct are clearly visible from the documentation, but the documented `From` impls can be hard to miss.

Maybe we save these changes for a followup PR to allow for further discussion (if any)?